tests/git.rs

```diff
@@ -71,6 +72,103 @@ fn cargo_compile_simple_git_dep() {
       execs().with_stdout("hello world\n"));
 }
 
+#[test]
+fn cargo_compile_forbird_git_httpsrepo_offline() {
+    use cargotest::ChannelChanger;
+
+    let p = project("need_remote_repo")
+        .file("Cargo.toml", r#"
+
+            [project]
+            name = "need_remote_repo"
+            version = "0.5.0"
+            authors = ["chabapok@example.com"]
+
+            [dependencies.dep1]
+            git = 'https://github.com/some_user/dep1.git'
+        "#)
+        .file("src/main.rs", "")
+        .build();
+
+
+    assert_that(p.cargo("build").masquerade_as_nightly_cargo().arg("-Zoffline"),
+                execs().with_status(101).
+                    with_stderr_does_not_contain("[UPDATING] git repository [..]").
+                    with_stderr_contains("\
+[..]Unable to fetch `Branch(\"master\")`: you are in the offline mode[..]")
```

Comment:
Could this also use `with_stderr` to match the entire output?

Also could the output here perhaps get improved? The debug representation may not be the best thing to print out and present to users here.